controllers/storagecluster/cephcluster.go

```diff
@@ -266,6 +266,10 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		return reconcile.Result{}, nil
 	}
 
+	if found.Status.CephStorage != nil {
+		sc.Status.DefaultCephDeviceClass = determineDefaultCephDeviceClass(found.Status.CephStorage.DeviceClasses, sc.Spec.ManagedResources.CephNonResilientPools.Enable, sc.Status.FailureDomainValues)
```

Comment:
Are we sure the `sc.Status.FailureDomainValues` will already be set at this point? If manual testing looks good, then it must be.

Reply:
Yes see here https://github.com/red-hat-storage/ocs-operator/blob/b18b234e14f6182c858e64bffd3e627b1a5350b3/controllers/storagecluster/reconcile.go#L393.
The failuredomain and such thing are set first before we reach cephcluster reconcilation.